src/components/Header/HeaderAction/HeaderAction.jsx

```diff
@@ -116,6 +116,7 @@ const HeaderAction = ({ item, index }) => {
       className={`${carbonPrefix}--header-action-btn`}
       key={`menu-item-${item.label}-global-${index}`}
       aria-label={item.label}
+      onClick={item.onClick || (() => {})}
```

Comment:
Could `onClick` be a non required prop on the `item` object with a noop as a defaultProp, instead of this conditional?

Reply:
(Header.jsx) the onClick prop itself is not required:
`export const HeaderActionItemPropTypes = {
 label: PropTypes.string.isRequired,
 hasHeaderPanel: PropTypes.bool,
 btnContent: PropTypes.node.isRequired,
 childContent: PropTypes.arrayOf(PropTypes.shape(ChildContentPropTypes)), 
 onClick: PropTypes.func,
};`
but it the entire prop array is made "required" in (HeaderAction.jsx):
`export const HeaderActionPropTypes = {
 item: PropTypes.shape(HeaderActionItemPropTypes).isRequired,
 index: PropTypes.number.isRequired,
};`

Reply:
Looking deeper at this - I think what you have is fine. There’s no way to set default props for a nested object without doing a deep merge comparison which is way more overhead than is necessary for this.